src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(toFileName, createReport(readFromFile(fromFileName)));
+    }
+
+    private String readFromFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(" ");
+                value = reader.readLine();
+            }
+            return stringBuilder.toString();
+        } catch (IOException e) {
+            throw new RuntimeException("Error: file not found: ", e);
+        }
+    }
+
+    private String createReport(String dataFromFile) {
+        int buyQuantity = 0;
+        int supplyQuantity = 0;
+        StringBuilder stringBuilder = new StringBuilder();
+        String[] split = dataFromFile.split(" ");
+        for (String arrays:split) {
+            if ("buy".equals(arrays.replaceAll("\\W\\d+", ""))) {
```

Comment:
```suggestion
        for (String record: records) {
            String splitedRecord = record.split(",");
            if ("buy".equals(splitedRecord[OPERATION_TYPE_INDEX])) {
```
lets make your code more efficient split just once instead of calling replaceAll 4 times

Reply:
Thanks for advise =)